File: solutions/javascript/difference-of-squares/1/difference-of-squares.js
export class Squares {
  constructor(n) {
    this.n = n;
  }

  get sumOfSquares() {

    return (this.n * (this.n + 1) * (2 * this.n + 1)) / 6;
  }

  get squareOfSum() {

    const sum = (this.n * (this.n + 1)) / 2;
    return sum ** 2;
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares;
  }
}

File: solutions/swift/difference-of-squares/1/Sources/DifferenceOfSquares/DifferenceOfSquares.swift
class Squares {
    private let number: Int

    init(_ number: Int) {
        self.number = number
    }


    var squareOfSum: Int {
        let sum = (number * (number + 1)) / 2
        return sum * sum
    }

  
    var sumOfSquares: Int {
        return (number * (number + 1) * (2 * number + 1)) / 6
    }


    var differenceOfSquares: Int {
        return squareOfSum - sumOfSquares
    }
}
